src/__internal__/focus-trap/focus-trap.component.js

```diff
@@ -53,7 +54,26 @@ const FocusTrap = ({
         setLastElement(elements[elements.length - 1]);
       }
     }
-  }, [children, hasNewInputs, wrapperRef]);
+  }, [hasNewInputs, wrapperRef]);
+
+  useEffect(() => {
+    const observer = new MutationObserver(function () {
+      updateFocusableElements();
+    });
+
```

Comment:
I think `const observer = new MutationObserver(updateFocusableElements);` should work

Reply:
Good point 😅